### DISCUSSION TITLE
Some questions about Style and Resource

### DISCUSSION THREAD

**Discussion by qqJoker** [AvaloniaUI/Avalonia#11316] [General]
When I first came into contact with AvaloniaUI, I referred to the documentation on the official website: https://docs.avavaloniaui.net/docs/styling/styles
So almost all the styles in my project use the Style method. But then I checked the source code and some open source UI style libraries, and found that they all use the Resource and ControlTheme styles. I have tried the above two methods, and the final effect is the same, but I don’t quite understand it. What is the difference between the two styles, or under what circumstances do we use Style and under what circumstances do we use ControlTheme?

In addition, when I was developing with the Rider tool, when writing ControlTheme in the Resource file, the compiler could not provide normal code completion prompts, which made me think that my code was wrong

**Comment by timunie** [comment]
If you are checking the source code, please keep in mind that the master branch is for the upcoming version 11.0 which has a bit different styling concepts. The docs are still for the current stable version 0.10.x. If you are using that version, please change the github branch to `stable/0.10.x`

![image](https://github.com/AvaloniaUI/Avalonia/assets/47110241/f61cdc1e-e18b-4240-a808-ef0fd82feded)

**Comment by robloo** [comment]
ControlTheme is brand new for Avalonia v11. It is a parent container for styles that encapsulates everything necessary for a single control. Now all the various pointerover, etc, pseudoclass styles as well as the control template and property setters are all styles themselves under a single ControlTheme. This ensures default styles no longer leak and allows completely switching out all related styles for a control together. This goes along with light/dark theme switching as well.
